### ISSUE TITLE
new module as remote

### ISSUE THREAD

**Issue by boelle** [basilfx/TRADFRI-Hacking#46]
hi

this project on thingiverse https://www.thingiverse.com/thing:3655354 has been much fun to me

it was easy to remove a zigbee module from their light bulbs and then reprogramm it to work as a 5 button remote

but now ikea is slowly replacing their zigbee modules with a new one that for the bulbs concern are far more power hungry and why should ikea bother as they are intended to be always on mains power

and their new 5 button remote do not have a module at all but is one complete pcb and to top it off they now use 2 AAA batteries

that makes it impossible to install it in the danish LK Fuga design, it will simply be to big to sit flush with the other switches

anyone that have started to look in to if the new modules can be reprogrammed to work as a remote and be energy efficient so that it can run on coin cells?

i emagine that the firmware needs to be taken from the new 5 botton remote and changed so that its as efficent as possible

of course the chip on the new module an the new 5 botton remote needs to be the same

and the pcb in the new remote is way to be to be but in a LK switch, from what i have seen online its 2 cm to big in each direction

**Comment by MattWestb** [comment]
The 3 gen devices is having one very good original Silabs module that is **not** battery hungry.
And its also possible making custom boards for the new 4 button remote (E2001/E2002) and not using the IKEA PCB if you like but its little more tricky to flashing then its having one newer CPU (need one real J-Tag probe).

And you can using the module from other 3 gen devices if you can flashing it (the E2001/E2002 is not debug locked so its possible dumping and flashing it).
![IMG_20210505_143614](https://user-images.githubusercontent.com/49618193/129255759-08692bd1-2ca3-47aa-8374-922d3f798e60.jpg)
(Dumped  and its have 5 buttons if counting the reset button.)

PS tuya is running 730 days on one CR2430 with one nearly the same module (ZS3L) with more LEDs on the PCB.

**Comment by boelle** [comment]
is that the new 5 botton remote? 

![image](https://user-images.githubusercontent.com/3832239/129258430-f37445fe-5a52-413b-a448-73829bc952ca.png)


in denmark i can only get it as a set with a bulb, but i think that will change

**Comment by boelle** [comment]
btw... i get the modules from 904.087.97

**Comment by MattWestb** [comment]
No its not the new 5 button remote then its one 4 button remote if not counting the reset button.
Its the new E2001/E2002 remote "styrbar" and the PCB is looking like this with one Silabs MG21 module.
![IMG_20210503_113654](https://user-images.githubusercontent.com/49618193/129259027-8ece6dae-5cc1-44e1-b472-f5da04fb21f5.jpg)
![IMG_20210503_113557](https://user-images.githubusercontent.com/49618193/129259060-742579bc-e892-4a19-9525-06cf946cd7fc.jpg)

Its very "clean" and shall being easy for the Danes making one updated PCB for that  is fitting in the original deigned switch.

Your module is one ICC-A-1 = the normal 2 gen module.

**Comment by boelle** [comment]
ahh

i wonder why they changed it to 2 AAA then? ie if the power consumption is the same

**Comment by CableCatDK** [comment]
I was told the new module would spend too much power. Why else is it using 2xAAA... Maybee I sould give it a try.
![image](https://user-images.githubusercontent.com/13363063/129259943-58af7c3e-88e5-4fb9-a311-e25ab3f84d17.png)

**Comment by MattWestb** [comment]
I can being that IKEA devs have not maximizing the sleep current but the module / chip is very low power.

The reason for the 2 AAA shall being re chargeable and is the new green way IKEA have starting and also all new device having only recycles repairer packing (all plastic is gone).

If its little hungry put 2 CR4530 in parallel on the new PCB (some tuya is having 2 CR2032 and was having one old MG1B module).

Thanks for the PCB layout !!

**Comment by boelle** [comment]
> If its little hungry put 2 CR4530 in parallel on the new PCB

remember that space is VERY limited

**Comment by boelle** [comment]
currently we use OpenOCD on a pi to flash firmware

can that be used with the new remote too?

**Comment by MattWestb** [comment]
If they have adding support for ARM Cortex®-M33 the last half year but i have not reading that have being done.

The module datasheet https://www.silabs.com/documents/public/data-sheets/mgm210l-datasheet.pdf

**Comment by boelle** [comment]
i just saw that you had wires soldered to the programming header and wondered how you dumped/flashed firmware

**Comment by MattWestb** [comment]
Main flash and data is dumped without problem with Silabs WSTK (it have integrated OBD J-Tag) and Segger J-flasher and the same made with tuya ZS3L modules on LIDL device but have loading the LIDL LED stripe controller with EZSP 6.1.0.0 and is using it as one coordinator in ZHA.
![IMG_20201204_140319](https://user-images.githubusercontent.com/49618193/129266386-ca3662e4-b605-414c-be35-8bee9f676483.jpg)
(before sounding my SWD and com cables)
Its haveing the  same chip and is also good calibrated and have one good implemented RF part (for being one Chinese module).

**Comment by boelle** [comment]
sorry for "stepping in a turd" here, but are there other ways to dumb the firmware?

using a raspberry pi right now with the current modules it seems costly to get a new programmer, but on the other hand i bet the one from silabs uses windows software, not that i mind linux but it could be nice to have everything on the same machine

so its a bit up in the air, nice to use the setup i have, but that means a different machine just to flash firmware, with a new setup i can use my main pc

**Comment by boelle** [comment]
being not so tech savy, i assume you use the debug connector

![image](https://user-images.githubusercontent.com/3832239/129477527-17993e49-30c2-4d58-af33-cf5b1657bec4.png)

we use the same currently

![image](https://user-images.githubusercontent.com/3832239/129477562-bb60c4f0-281d-467b-a8e4-b04ba84ed370.png)

but not sure if that would work, i do not have a one of the new modules, but @CableCatDK does

**Comment by MattWestb** [comment]
With one J-Link (every original version is working also the EDU) and you can building one "OBD clone" from one STM32 board and using it in windows but dont letting Segger updating the firmware on it then its being soft broken. Using Segger flasher and commands programs is working OK for dumping but can being it dont like flashing if having one clone or ST-Link flasher (that is ST OBD flasher only allowed flashing ST chips). 
But flashing can being made with Silabs Commander that dont care if its one clone or not (its free downloading it from Silabs web).

The pins used for flashing is the same SWD for both 1 and 2 gen modules only changed is the CPU that must being supported by the probe and / or flashing software.

**Comment by boelle** [comment]
So for the danish dummie here it would make more sense to just get the WSTK programmer?

all i'm after is being able to dumb firmware from the 4 button remote and then flash it again to the same module. Thoase module i can get from mouser.

**Comment by MattWestb** [comment]
The WSTK is expensive but then you also is getting access to Silabs Zigbee stack SDK and one good thing is that with the WSTK you can also unbrick "hard software bricked devices" (boot loader flashed at wrong address with Silabs commander) that is not possible dong with other probes (i have bricking 3 IKEA modules and recovering them with my WSTK).

The cheapest is one Chinese ODB clone but dont have any grantee that is working so long (then Segger is blocking the serial numbers of them) but one Segger EDU is not so expensive and working very well and is much smaller then one WSTK.

Then dumping the flash is the main flash and also the user data that is needed then its can (but not always) having configuration for the firmware that making it working in one different way.

**Comment by CableCatDK** [comment]
First release of STYRBAR to Danish wall switch is now released:
https://www.thingiverse.com/thing:4938961
The idle power consumption is the same.
![IMG_4337](https://user-images.githubusercontent.com/13363063/135489948-0551a4b2-e6ff-4c10-bb0f-40ba2bead3d2.jpg)

**Comment by CableCatDK** [comment]
Diagram
![image](https://user-images.githubusercontent.com/13363063/135490215-dda723a2-d522-4a2e-a716-8547b374ba65.png)

**Comment by MattWestb** [comment]
Very nice !!!

I think the Zigbee module is not so easy getting in the open market the IKEA, tuya Philips and many more is having high demand for it and Silabs cant producing enough of it but the have opening and certifying more productions cites the last year (if reading they internal company PM that is also distributed to devs).

Our (community) firmware cooker Gary R have ordering some hundred modules for nearly one year go and cant getting  them shipped then they is out of stock before the shipping is arriving to the seller.

The cheapest device i can getting in Wien is the same new E14 or E27 you have used for 8€ and need reprogramming the module.

But i was 2 week in southern Spain for holiday and doing one "pi stop" at IKEA in Montpelier and was baying 2 stainless steel version Styrbar for only 10€ and in most countries they is costing 15€.

Great that the power consumption is OK and not killing the battery and also possible putting one large inside if needed.

Thanks for the schematics it great to having if doing one other version of one wall switch.

For dumping the firmware you can finding one version with GDB in my git but is also possible using Sergger Commander and also Silabs Commander but the last is not so easy like using the GUI from Segger Commander for normal users.

One more time great work done Per !!!

**Comment by CableCatDK** [comment]
In Denmark STYRBAR is 69 DKK, whis is 9,28 euro. The nye E14 is 59 DKK, which includes the new module. The module alone is about 100 DKK.

Reprogramming the module (in Danish with nice pictures): https://rbx.dk/w/smart-home-knapper-og-paneler/flash-ikea-zigbee-j-link.html

**Comment by MattWestb** [comment]
For my "Billy EZSP" project i was baying the cheapest "family pack" for 10€ with one E27 WW (for the module) and E1744 remote and its was very useful.

So better baying the _nye_ clear E14 and getting little spare parts for the lab ;-))

I think IKEA was having one very hard to getting the good deal with Silabs (or it was Silabs that was having the problem getting the deal) then some FCC papers is over 2 year old with the MG21 module and they was having them on hold then was not getting one good price from Silabs but its very normal for IKEA if not getting one good deal then is normally leaving the project and taking the costs (and some time the manufacturing company is going bankruptcy then not having any large customer like one Swedish bed manufacture).

Its also looks like the SYMFONISK shall have getting one MG21 module but was getting one ICC-1-A in the end.

So very likely many Zigbee modules from IKEA lights is going to the DIY and professional projects but is not so bad !!

The latest is IKEA have ordering containers then they cant getting the products to the market then its shortages of them (Maersk ?) and is also looking baying own ships for going around the time and price of the shipping from China.

One question can you reusing the 5 switches from the Styrbar PCB to your custom PCB ?
Then its not so bad baying one Styrbar then can reusing most of the components.

**Comment by CableCatDK** [comment]
The TACH switches are different on my PCB.

**Comment by CableCatDK** [comment]
The SONOS gen2 is very similar to STYRBAR. IT has 3 more buttons, connected to PC02, PC03 and PC04.
![image](https://user-images.githubusercontent.com/13363063/221275949-dd0329f9-3b64-43b2-a9e6-b793d2c99545.png)

**Comment by MattWestb** [comment]
Is it Synfonisk 2 ??
One user have getting one in thee Netherlands and in Vienna it shall coming the next week.
I  dint knowing what commands is suing but its having very normal cluster for one light controller but we is knowing mor in some days :-))

**Comment by CableCatDK** [comment]
You should watch this video I made: https://youtu.be/msPRU2MsIXY

Here are the functions, when pair directly with a lightbulb:
![image](https://user-images.githubusercontent.com/13363063/221282176-18cc715d-4142-4694-aaf1-70e000f48cb1.png)

**Comment by MattWestb** [comment]
### Nice !! 
So normal OnOff and Level commands plus 4 more functions IKEA have cocked in the firmware.
New version with one AAA of OnOff, ShirtCut and OpenClose is in the pipe (FCC certed).

The reset / sync of light is long press right or left button on Styrbar.

**Comment by MattWestb** [comment]
By the way OnOff and OpenClose is using the same firmware and you can dumping the flash aria "user data" and flashing it on the other and you have transforming it.

**Comment by CableCatDK** [comment]
> By the way OnOff and OpenClose is using the same firmware

Very interesting. Can one change the user data over the air?

**Comment by HexDK** [comment]
> By the way OnOff and OpenClose is using the same firmware and you can dumping the flash aria "user data" and flashing it on the other and you have transforming it.

you mean this?

![CleanShot 2023-02-24 at 22 08 21@2x](https://user-images.githubusercontent.com/415476/221293007-2182b185-a4aa-42f4-a8a7-ba1dfeae9246.png)

**Comment by MattWestb** [comment]
Yes but Styrrbar is using the new module so its not working with it.
E1743 OnOff is here [E1743UD.zip](https://github.com/basilfx/TRADFRI-Hacking/files/10828826/E1743UD.zip) but i dont have the OpenClose in my laptop but dumping it and you can transforming between the device.

**Comment by MattWestb** [comment]
The interesting is not the name (is being used on the basic cluster) but the other bits after it is changing hardware functions in the firmware.

**Comment by HexDK** [comment]
> Det interessante er ikke navnet (der bliver brugt på basisklyngen), men de andre bits efter det er at ændre hardwarefunktioner i firmwaren.

it not all version they use userdata maybe

![CleanShot 2023-02-24 at 22 29 02@2x](https://user-images.githubusercontent.com/415476/221297036-336af25c-0b33-4d41-bc30-0923f9239772.png)

**Comment by MattWestb** [comment]
If not having any custom data in `user data` the firmware is using the default its having (its in the end of the firmware file around "IKEA").

Its the same with the lights is many having the same firmware and its one model in it but all other versions is getting it thru `user data`.

The 5 button is little different then its old and new hardware but the same firmware.

**Comment by CableCatDK** [comment]
### I have finished the diagram

Changes from STYRBAR to SONOS gen2:
- LED moved from PC01 to PC00, which was wired to VCC. This makes layout easier, and frees up PC01.
- CLK and DIO are not longer used for buttons. Which interfered programing (huge improvement). 

![image](https://user-images.githubusercontent.com/13363063/221351289-c76beba6-9ed5-4ae6-a825-ffecd4a85774.png)

**Comment by MattWestb** [comment]
### Looks great !!
I was just by Westbahnhof and have getting 2 of them :-))
I need sniffing how IKEA is setting them up for getting the extra keys working and implanting it in ZHA.

Try flashing the dumped Symfonisk on one MG21 module and look is its working OK.
If not i think you also need flashing the userdata for getting it working but if its working OK  then its not needed doing the work.

Devices that is using the debug pins for hardware is very bad then trying flashing / dumping and can making it  impossible getting  it working. 

Next 3 devices that is in the pipe https://fccid.io/FHO-E2201/Letter/Model-Declaration-Letter-6321956.iframe.

**Comment by CableCatDK** [comment]
I looks like IKEA will EoL the ON/OFF, curtain and shortcut button, with these 3 new devices.

**Comment by MattWestb** [comment]
I think the old Zigbee  module is going  complete out and also the FER32MG1P chip is end of life.
Then looking one  the first EFR32MG21 devices on FCC it was over one year of the drawings before IKEA was applying for FCC so i think they was having one very long and intense round with Silabs for getting one very good price on the new module and also Silabs have open 2 new assembly lines for the  module so they can delivering it to IKEA.

By the way the module is engendered in Finland !!

And with Diriigera is the Symfonisk 2 getting firmware update from 1.0.012 to ???
Is tradfri doing the same ??

**Comment by MattWestb** [comment]
Updated  firmware is version 1.0.32 for the E2123.
Sniffed then Dirigera was doing the updated and rebuild with zigpy tools if someone need it (the signing is OK so iit shall being 110% OK). 
[ota_t0x110e_m0x117c_v0x01000032.zip](https://github.com/basilfx/TRADFRI-Hacking/files/10831525/ota_t0x110e_m0x117c_v0x01000032.zip)

Have some one that  have opening the device dumping the user data ?
Im interesting of it for future use.

**Comment by HexDK** [comment]
maybe I'll go to Ikea(Aalborg) tomorrow and pick one up so I can dump the firmware from it 
if Ikea has more in stock

**Comment by MattWestb** [comment]
If you is getting one i very thankful getting the dump.
I must getting the support for it in ZHA and also trying sniffing and decrypting the OTA server function Dirigera is using than its not the same feed as TF.

Westbahnhof in Vienna was only getting one package with 20 and i was baying the first pair from them so its not large quantity they is have getting but its needed then the first gen is out of stock for long time.

Bay the way the OTA file looks very different form all other ones with very mush c-code in clear and references.

**Comment by MattWestb** [comment]
O i was forgetting IKEA is open on Sundays like in my Xhome country (Swe) !!

Is someone having or knowing one that have KNYCKLAN Open/Close remote (E1841) and the KNYCKLAN Receiver (E1842) that i cant getting in Austria but like have firmware dumps off (the remote is most interesting then its very likely having the same hardware ad ObOff / OpenClose ones) ?

**Comment by CableCatDK** [comment]
When I try to dump the flash, I get this error:
```
WARNING: Could not connect to target device
ERROR: Debug access is locked. Could not connect to device
```

Device Info from Simplicity Commander
![image](https://user-images.githubusercontent.com/13363063/221372674-b7aed36b-dbb5-406a-80c2-21bf4b4e52c7.png)

**Comment by MattWestb** [comment]
Ops IKEA have locking the debug and start using secure elements !!

So you can **only** doing one device erase and **not** reading / writing the device.
The good is they is not using secure boot = the bootloader is using encryption with  keys stored i the secure storage.

If you like trying flashing the sniffed OTA firmware i was doing on one MG21 module with Styrbar bootloader and system and look if its liking it or not.
I think of 2 alternative: 1 the bootloader and / or the app is not liking it and is caching / locking the device 2 its working OK but can being  little strange device name from the styrbar.

If you doing one device erase you is losing the main flash and i think its also erasing the user data but im not 110% sure (need reading the bootloader paper).

I hope IKEA have not implanting the tamper functionality then it can burning the chip and cant recovering it at all.

**Comment by MattWestb** [comment]
Have extracting the signed OTA file and making one S37 file of  it so it shall being possible flashing it with commander and dont need converting it and its doing it in the right place.

Have putting in the GBL and BIN file 2 but the s37 is the safest / best for flashing.
[110e.zip](https://github.com/basilfx/TRADFRI-Hacking/files/10832225/110e.zip)

**Comment by CableCatDK** [comment]
So should I try flashing a blank module, and solder it in place instead of the current module, to see if we can make your own module?

**Comment by MattWestb** [comment]
Take one not used module then you cant flash the Symfonisk, you can only erasing the chip = you have loosing 20€ and after that flashing it but perhaps the Symfonisk firmware.

New module flashing one dumped Styrbar firmware from 0x0 so you is getting its bootloader and APP.
Then flash the 110e.s37 over it so its having the standard  bootloader and the Styrbar APP.

If the bootloader is OK and the APP is loaded OK and liking all things then it shall working.
If have make more protection its not loading OK like putting keys in the `userdata` that we cant reading.

**Comment by CableCatDK** [comment]
I have flashed 110e.s37 to one of my converted STRBAR to FUGA. The result is that the led is half bright constantly, and nothing else work. This result is expected, as the pairing button has been changed to pull to VCC, instead of pull to GND, and the LED output has been changed.

Flashing the previous firmware, recovered the module.

I have not tested it yet in the SYMFONISK PCB. I need a donor module. Maybe one of my noisy RGB bulbs will get the hammer.

**Comment by MattWestb** [comment]
I hope Synfonisk FW can running on the Styrbar bootloader but only way to knowing is flashing as you have doing and testing and if you is having one PCB with all pins OK its being easier testing.

Before doing chip erase of the CWS3 can you dumping main flash and userdata ?
I think it can being great making one RGB LED controller that  is good Zigbee 3 compatible.

You can also testing if Synfonisk is starting OK on the CWS3 bootloader with only flashing the OTA files. And one test thing is deleting the userdata on the new module so its not somthing that making it crashing as one second way to testing

**Comment by CableCatDK** [comment]
I just brought a 204.867.84 as a donor chip. But it is locked too. I tried to lock the module. But I think I have bricked it.

**Comment by MattWestb** [comment]
With Silabs commander is shall being possible unlocking the debug lock in the chip but then its erasing the internal flash. I have not trying it but shall working OK.

I only have hard software braking 3 old IKEA modules but was debricking them with commander CLI but its need one original Silabs WSDK that have special macros implanted but the debug unlocking shall working with normal Segger J-Link probe and commander.

Look in the manual for Silabs Commander how its shall working.

So we have making / forcing IKEA debug locking there modules by using them for nice things wall light switches and Zigbe coordinators and Open Thread Boarder Router RCPs !!!

Bad swedes !! ;-))

**Comment by MattWestb** [comment]
Also make sure connecting the reset pin then commander have updating Segger libs and have problem sometimes (not always) if the reset pin is not connected and can making very strange things without it.

**Comment by CableCatDK** [comment]
I would like to recover the module. But for now, the high-pitch RGB-bulb gets the hammer tomorrow.
I have finish my first revision of the PCB: https://www.printables.com/model/410259

**Comment by CableCatDK** [comment]
Erasing the module can be done with:
```
c:\Simplicity Commander>commander device lock --debug disable -s seriel-of-segger -d MGM210L022JNF2
Unlocking debug access (triggers a mass erase)...
Chip successfully unlocked.
DONE

c:\Simplicity Commander>commander device unlock -s seriel-of-segger -d MGM210L022JNF2
Unlocking debug access (triggers a mass erase)...
Chip successfully unlocked.
DONE
```

**Comment by CableCatDK** [comment]
When adding the STYRBAR moded to SYMFONISK. It appers as a STYRBAR and 2 shortcut buttons:
![image](https://user-images.githubusercontent.com/13363063/221523616-90eae092-aae4-4857-a8ab-5c8cac2b70d5.png)

**Comment by MattWestb** [comment]
### That is great !!!
I was reading yesterday evening that  is need one Silabs WSDK for doing recovery and unlocking of devices but  then its working for you :-))
Wot SWD adapter is you using for flashing ?

The Styrbar is only having one normal endpoint and Symfonisk 2 is having 2  extra with name `shortcut 1` and `2` that is sending the commands from the extra buttons.
If you is using one open Zigbee system you need binding the 2 extra endpoints to the coordinator so its sending the commands to it.

If adding it to one IKEA system TF or Dirigera it can being confusing and not working OK but im interested if its relay working but i think so then the APP / firmware have starting OK and can being added.

Then the OTA file i having the name in it you can trying flashing one empty file or erasing the flash pagers of the `userdata` so its not using the old name.
Fromm the OTA fiile:
```
IKEA of Sweden
SYMFONISK sound remote 2
20221219
E2123
1.0.32
```
Is you using TF for your system or do you using other HA like ones ??

So in the end i was not getting any dump pf the  CWS3 ;-((

**Comment by HexDK** [comment]
STYRBAR moded to SYMFONISK. Test Version
[Test_Version.zip](https://github.com/basilfx/TRADFRI-Hacking/files/10841142/Test_Version.zip)

**Comment by MattWestb** [comment]
@HexDK Have you hex edit Styrbar userdata or making debug unlock and dumping it ?
Im not sure if debug unlock is easing the userdata then its little defuse in the docks (zherfor i have not recommending trying  it before knowing if Styrbar bootloader is working for not bricking the device.

If its working then is IKEA chested then can writing main flash with bootloaader and OTA APP and also the uerdat and also "recycling" (debug lock unlocking) old modules and using them for Symfonisk 2.

Only problem is if IKEA have making some extra in the in the Symfonisk bootloader so further OTA is not working. The good thing is the module is the with software secure wault and not the hardware ones that cant being workaround like this.

**Comment by HexDK** [comment]
Yes I have HEX edit user data. and when you unlock I think it deletes user data, but I have not tested. don't have one that was locked

**Comment by MattWestb** [comment]
Good !
Normal erase flash chip is not deleting the userdata but debuglock unlocking shall erasing all and also the internal RAM so cant sniffing secret manufacture data.

If you like you can locking the debug and unlocking it and dumping the userdata then we is knowing.

**Comment by HexDK** [comment]
yes, I want to do that test tomorrow, I'm just at work right now :)

**Comment by HexDK** [comment]
![image](https://user-images.githubusercontent.com/415476/221651860-21599438-86ae-42ca-8102-ca0016a3b856.png)

**Comment by CableCatDK** [comment]
Changing userdata worked. The remote is now proper recognized as a SYMFONISK remote in the hub-

NAME = SYMFONISK sound remote 2
MODEL = E2123

```
S3150FE0020053594D464F4E49534B20736F756E6420CD
S3150FE0021072656D6F74652032FFFFFFFFFFFFFFFF13
S3150FE00230014532313233FFFFFFFFFFFFFFFFFFFFC5
```

**Comment by CableCatDK** [comment]
Here are both the source files, and the result:
[SYMFONISK_E2123_firmware.zip](https://github.com/basilfx/TRADFRI-Hacking/files/10843862/SYMFONISK_E2123_firmware.zip)

**Comment by MattWestb** [comment]
I was thinking recommending using s37 files for doing the update so can getting all in one file and in right place but im to late ;-(.

Is the  `IKEA_TRADFRI_SYMFONISK_sound_remote_2_E2123_1.0.32_mainflash_userdata.s37` flashing all on one blank module so not need doing any more steps or is it needing more flashing ?

**Comment by CableCatDK** [comment]
> Is the  `IKEA_TRADFRI_SYMFONISK_sound_remote_2_E2123_1.0.32_mainflash_userdata.s37` flashing all on one blank module so not need doing any more steps or is it needing more flashing ?

The file have the entire mainflash and userdata. So the module can be flashed in one step. Or 3 steps, if you need to unlock it first.

**Comment by MattWestb** [comment]
Great !!
I shall testing it.
Was testing locking one Billy (old IKEA module) and unlocking it but was not working.
Must using the Silabs recovery for getting it back:
```
.\commander.exe  device lock --debug disable --speed 100 -d efr32mg1p
Setting debug interface speed to 100 kHz
Unlocking debug access (triggers a mass erase)...
Unlocking debug access (triggers a mass erase)...
Attempting EFR32/EFM32 Series 1 recovery using system bus stall...
Attempting connection while holding device in reset...
Waiting for AAP mass erase to complete...
Verifying debug access...
Found DP ID:  0x2ba01477
Found AAP ID: 0x24770011
Chip successfully unlocked.
DONE


.\commander.exe  device masserase  -d efr32mg1p
Reconfiguring debug connection with detected device part number: EFR32MG1P132F256IM32
Erasing chip...
Flash was erased successfully
DONE
```
The chip was having bootloader EZSP NCP APP and user data set before unlocking.
After recovery / unlocking it was having the main flash blank but **not** the user data.

Was testing  one Markus module (IKEA new EFR32MG210L module) that was having bootloader,  RCP (Open thread broader router radio co possessor firmware) and model sett in user data. Debug locking and unlocking with mass erasing and  the main flash is blank but the user data is **not** !!

Have some brave ones testing unlocking the debug lock on the original module ??
I think it very likely not erasing the userdata and can being dumped and reused as one original devices on our recycled modules.

If getting the original userdat its no way IKEA can making problems in the future then they is not using security storage in the chip so all is in the main flash and userdata.
Then OTA updates in the future shall is working for 110% !!

**Comment by MattWestb** [comment]
I was flashing the combined and taking all cables away and only VCC  and GND (I using PC05 ass comport so cant having the debug connected) and then using one resistor between VCC and PC05 and its start requesting beacon. Adding it in Dirigera and i is getting 3 lights with the model from the origin of the module but the "naming" is remote X.

So the firmware is working but somthing its not OK userdata.
I think its only updating the changed bits and not erasing the not changed ones but i must looking if its so or not.

**Comment by MattWestb** [comment]
Userdata dumper and its not updated with the combined s37 file :-((
[SyFo2.zip](https://github.com/basilfx/TRADFRI-Hacking/files/10850250/SyFo2.zip)

**Comment by CableCatDK** [comment]
I have learnt that this command:
```
commander.exe readmem -s seriel-of-segger -d MGM210L022JNF2 --region @mainflash @userdata -o IKEA_TRADFRI_SYMFONISK_sound_remote_2_E2123_1.0.32_mainflash_userdata.s37
```
Does not dump both mainflash and userdata. But only mainflash.

New fixed firmware is here:
[IKEA_E2123_1.0.32_firmware.zip](https://github.com/basilfx/TRADFRI-Hacking/files/10852756/IKEA_E2123_1.0.32_firmware.zip)

**Comment by MattWestb** [comment]
Thanks !!
I using one Silabs WSDK and its have internal module or external device and if using the external i must using `--device` for the chip i have and then the `--region @userdata` is not working and must using range but it working OK but not so easy a region.

I think only some days and we have working around IKEAs bad protecting there new devices is great !!

And you must ordering new PCBs for you new wall switch :-))

I waiting Vindstyrka air sensor shall being released and the OTA file for it and then trying using Styrbar as ground for getting it working on new Zigbee module and also Starkvind i like trying but not so easy finding the paring pin and if its grounding or VCC that is needed and can burning the chip if doing it wrong. Also if holding it too long its tush linking other device.

**Comment by MattWestb** [comment]
I have [zigpc-cli](https://github.com/zigpy/zigpy-cli#dump-embedded-firmware-for-further-analysis) installed in one python env in windows and using the command `zigpy ota dump-firmware .\10082261-zingo_lds_starkvind-1.1.001.ota.ota.signed E2006.gbl`. You is not knowing if its one old EBL or one new GBL file but can using silabs commander for getting information of the file. Then using commander for converting it to S37 or other format like bin, hex, ebl or gbl files. Also combining files to one hex or s37 is working OK.

For info of the OTA file `zigpy ota info <file>`.
The command `zigpy ota reconstruct-from-pcaps` is not implanted (its work in  progress) but its what i have using making the OTA file from sniffing the Zibee network then the device was being updated.

**Comment by MattWestb** [comment]
I was looking yesterday in your git and great collection of dumps that is always good to have / knowing where to finding :-))
I normally doing s37 files then its easier to flashing and dont need putting the right address every time.

In commander then clicking on read memory you is getting one nice grafik.

With the updated firmware its little strange but its working.
First is the bootloader 0x0 to 0x3ffff and then the APP at 0c40000 to its end. Then the APP is starting its making one NVM3 in the last part of the flash. The dump have some extra blocks allocated but its looks working OK (can being rests from the Sytrbar NVM2 file).

Silabs commander is working OK but it little trickey and the documentation is not 100%.

I think all is very fast learning and in the end we have getting all working well !!

**Comment by HexDK** [comment]
We can hope that dump we can make when it is again updated via ikea hub will be a better dump

**Comment by MattWestb** [comment]
@CableCatDK I was looking on the schematics and i think paring switch / PC05 is having the pull up resistor wrong. 
PC05 is having R8 as pull up to VCC and on the other side the switch its connected to VCC.
For it to working OK it shall being pull down to GND on the PC05 side of the switch and VCC on the other side so PC05 is normal low and is getting high the pushing the switch.

If having PC05 to VCC all the time its very likely the device is trying tush link devices all the time if the program is not have blocking it then the switch is continuous closed.

On my test Symfonis 2 i have it not connected and only putting it to VCC with one resistor for paring and its working but its not by the book.

I have repairing 3 signal repeaters and 2 light and 6 sensors that was by mistake tush linked of the remote then testing and hope no more devices have being **kidnapped** of my wild testing.

**Comment by CableCatDK** [comment]
Corrections:
-  R8 (unpopulated) is between PC05 and GND, not PA03.
- C8 renamed to C4, is between PC05 and VCC, not GND (need to be fixed in manufactured PCB).
- C9 renamed to C1.
- C10 renamed to C2.
- C6 renamed to C8.
- C5 renamed to C10.
- C3 renamed to C7.
- C4 renamed to C9.
- C7 renamed to C3.
- C1 renamed to C5.
- C2 renamed to C6.
- SW1-7 are between their IO-pin and PA03, not GND (mayor flaw, will take a lot of fixing).

![image](https://user-images.githubusercontent.com/13363063/222175670-04dc975d-4731-40d4-a1a8-34df2bd80ae6.png)

**Comment by MattWestb** [comment]
That looks more logic to my.
So the PA03 is normal high and if some button (not the paring one) is pressed its going  low and waking the SOC up for scanning with key is pressed and sending the command from the key and all caps is for filtering the contacts on the switches.

Is C4 populated and not R8 on the original IKEA PCB ?
I was only thinking **if** the C4 is leaking it can bringing the PC05 high and need R8 for being kept low for not going in pairing mode without pressing the pairing button.

If the device is going in torch link or normal paring mode then is being powered or without some reason i think populating one 1M as R8 shall fixing it.

I also think IKEA have putting pull up and down on the IO pins in the software and not need all resistors they have on the PCB (thinking = not knowing) or.

**Comment by CableCatDK** [comment]
I am working on a 1½ module version too. This is the button configuration I am going for:
![image](https://user-images.githubusercontent.com/13363063/222896338-d64c284c-e28e-4842-9336-6d0324de9d98.png)

**Comment by CableCatDK** [comment]
I have made ny first prototype.

The LED and pairing button are working. But not any of the other buttons. I have fixed the errors I described earlier. I am missing R9, maybe that is the issue.

![image](https://user-images.githubusercontent.com/13363063/223206410-a9a0d756-33e9-48b8-9c60-0fd29ff58dfe.jpeg)

**Comment by MattWestb** [comment]
**If** the inputs have internal pull down it shall working but i think R9 is needed and if not heaving SMD version use normal one for testing.
Its looks title as _Modern Art_ but the most importing see if its working and doing the correction in the design.

My "Symfonisk 2" is still running OK on the IKEA module so i think the file you is having shall working OK on all new modules only need flashing the userdata so its not having old crap saved in it that making strange things.

I was getting one VINDSTYRKA today and shall testing dumping it if its not debug  logged then its have date code 2236 then i hope getting the userdata and also one newer bootloader that is not locked of the firmware that we can using then IKEA is starting shipping all new devices with debug locked.

Next is being the upcoming OnOff, OpenClose and shortcut 2.

By the way the updated firmware (that you was getting) Symfonisk 2 shortcut cluster is sending Matter formatted commands and is only missing some small parts but i think IKEA like changing the cluster to the matter one (then its OK for Zigbee) and piping to matter in the GW.

**Comment by CableCatDK** [comment]
I am using the original module I desoldered earlier. I will see if I can get a 1M resistor.

![image](https://user-images.githubusercontent.com/13363063/223359505-944c1a5b-f06b-48d2-83ce-a5372965b26f.jpeg)

**Comment by MattWestb** [comment]
You can using one lower value for it then testing but for production its better with higher so your battery is not going out to fast.

One design question. 
All switches is having one cap over the switch.
Is it not better putting the caps between the legs of the switch then its enough space for it and only need small pads for it and its the same plane as the switch legs on both side and its making more free space on the rest of the PCB but only little more tricky soldering the components ?

In the end i like getting more metal free space around the Zigbee module antenna so the RF is not being killed and you is getting connecting problems on your devices.

The antenna free zone and guides is in the module data chest.

**Comment by CableCatDK** [comment]
Remember when I design the PCB, I thought the connection for the switches and caps where not the same.

This is the current 1,5M version:
![image](https://user-images.githubusercontent.com/13363063/223376557-ef40a9ac-ffa6-4f77-98b2-9035b7e3b473.png)

And the new 1M version:
![image](https://user-images.githubusercontent.com/13363063/223378129-ce2f6587-3662-4bf5-85bc-5d735f0f83ca.png)

**Comment by MattWestb** [comment]
The 1½ module version i think the antenna is better then the 1 module layout but you never knowing before testing it in real life (its radio waves magic).
PCB need the switches in that position and also the battery holder and the space for mounting so not so mush can being done.
Ops i think you is having 2 batteries on the large module :-)))

One bad new is that VINDSTYRKA cant being read with Commander then IKEA is using pins for other things like IC for LCD drive and sensors but with command line its saying debug locked and Seggers J-Flasher is saying the same so we is not getting one new bootloader for controllers (I think lights is having one other then have reset with power cycles) and updated firmware is certified but Dirigera dont have it but its coming later. Have putting in one ZHA test system without IKEA update so i can sniffing it then its being made and making one OTA file in the future.

So Styrbar we have one working bootloader that is working with Symfonisk 2 OTA files.
Looks like all new device is being debug locked also lights.
So likely the new upcoming buttons also is and hoping that Styrbars bootloader is working on them  or we is busted making _extending hardware_ for them in one easy way.

**Comment by CableCatDK** [comment]
It is Worning with a 1M resistor. 10K did not.

![image](https://user-images.githubusercontent.com/13363063/224304555-f26e4f07-d144-4721-990d-ce954734110d.jpeg)

**Comment by MattWestb** [comment]
Was testing little more and with x-modem bootloader and the update = not starting (is not making NVM file at the end of the flash).
Styrbar bootloader and update on erased chip not starting  (no NVM is made = chip is not starting).
So was using the normal patched and was writing copy of empty UD on flash pages that is wrong from some old stuff and looking in the flash map and the bad pages is erased OK.
Starting the chip and holding the pairing button pin high and its scanning tush link = working.
Was flashing one blank user data and testing and its starting OK.

Was paring with fixed main flash and empty user data with Dirigera and its looks very nice.

I hope IKEA have not saving some importing things in the user data and that the device is working OK in hard or software way and also that the future OTA is working with it.
I shall testing if its requesting OTA with ZHA OK then i think its no problems but that we is not knowing until the next OTA update is installed and working on the device.

[E2123.zip](https://github.com/basilfx/TRADFRI-Hacking/files/11001563/E2123.zip)

**Comment by CableCatDK** [comment]
The hammer falls:

![image](https://user-images.githubusercontent.com/13363063/226695163-9bc98f97-8143-48f6-b931-91be3319e37a.jpeg)

**Comment by CableCatDK** [comment]
Firmware dump and high res pictures of TRADFRI bulb E27 CWS 806lm
[LED1924G9.zip](https://github.com/basilfx/TRADFRI-Hacking/files/11033400/LED1924G9.zip)

![image](https://user-images.githubusercontent.com/13363063/226735311-4e6a8c75-eb10-4adf-b413-dfe692587e4b.png)

**Comment by MattWestb** [comment]
Now i understand way IKEAs CWS3 is mush cheaper then HUE lights they is not putting in all components like Zigbee modules and power supplies in China and IKEA can selling them for only some € ;-))).

The main flash is not old then the date is 2019.07.19 so the bootloader can being used for devices with debug log and rolling one OTA file on it after doing one unlock.

And the CSW3 is one true RGBWW but i think they is also using  the RGB for doing  the color temperature for helping then 20 W type white LEDs and only 3 G type white LEDs of the other CT i think its little unbalanced but can being that the G type is mush more powerful or they is using the RGB LED for doing compensation for getting it looks OK.

And thanks for the instruction it look great and i always forgetting how to do all parameters and must using the commander papers for doing it right.

And now you is knowing way many firs and second gen IKEA lights is having better radio performance then 3 gen devices.
The first and second gen is the MG1P one +19 dBm chip and the antenna is normally good places and 3 gen is MGM210L with 12.5 dBm and the antenna is RF grounded of the metal socket and is surrounding of metal all over for getting it smaller for visual design and destroying one very good implanted Zigbee modules performance.

Thanks for sharing the knowledge !!!

**Comment by CableCatDK** [comment]
Well version 1.0.032 does not work with IKEAs gateway. Is there anyway to get the orginal relase firmware 1.0.012?

![IMG_0580](https://user-images.githubusercontent.com/13363063/229367032-ec862566-06c1-4d6c-a807-6980e550a869.JPG)

**Comment by MattWestb** [comment]
We cant dumping the main flash then its debug locked so not possible.
What is not working the paring or the function of the device ?
I can move my updated devices to TF GW and looking if my is working that is original and updated but i have not seen that TF have putting it on the supported list in the release notes so it can its not supported in TF GW.

**Comment by MattWestb** [comment]
It was paring easily with TF GW and is shoeing Version 1.0.32 and no updated.
I dont have any Sonos speakers so i cant testing how its working.

**Comment by CableCatDK** [comment]
While pairing, the app says that parring failed. But the device is added in 2 groups (each of the shortcut button), but the main device is missing.

![IMG_6985](https://user-images.githubusercontent.com/13363063/229367752-7df599ed-f08a-4f3c-bf09-acc99e082d21.jpg)

**Comment by CableCatDK** [comment]
There is no secret download link for version 1.0.012= What is the link for 1.0.032? I there a JONS with download links for DIRIGERA, like for the gateway?

**Comment by MattWestb** [comment]
The GIRIGERA is using encrypted communication with the server and we have not braking it.
I have sniffing the Zigbee network then it was updating my device and was putting all packages together and the signing is OK so all is OK and the device is also flashing it that is not doing if the signing is broken.
All "found" OTA files and links iis posted heer  https://github.com/zigpy/zigpy/discussions/660 (DIRIGERA OTA files is on the right column and clicking on it for downloading it).

I was having the same problem with "extra devices" in DIRIGERA then not erasing the user data and only flashing the main flash you was doing and was writing the model of the light the module was build in before removed. So its somthing that is not OK in the user data that the firmware need.
You can trying writing one blank user data so its not getting any extra info but it can also its missing somthing that we need having in it.
If we cant fixing it we need doing one debug unlock on one original device and hoping it is nit erasing the user data and dumping it so we is having on original for our devices.

**Comment by CableCatDK** [comment]
Just to clarify. I have the samme issue, when I used an original remote. Updated via DIRIGERA, and then joined it to the gateway.

**Comment by MattWestb** [comment]
If its one original that is updated with Dirigera its the same status of the device like my updated one.
Then i think the manufacture have doing somthing wrong with  the data setting in the factory and writing false data in the user data (its not the first time its have happening with the OpenClose and some lights before).

But is the looking OK then pairing it with Dirigera ?

All IKEA controller i have looking on is blocking firmware download so we cant downgrading it also if we was having the OTA file with OTA if we was having it but we can extracting the GBL file and flashing it with SWD (on one unlocked module).

**Comment by MattWestb** [comment]
In TF GW i have one remote under sound and 2 shortcut buttons shortcut buttons and you is have getting the last 2 as somthing else that is one normal light controller then the GW have making rooms = Zigbee groups for the 2 extra endpoints for the shortcut buttons.
The name and functions of the 2 extra EP is stored in the user data of the device.

**Comment by MattWestb** [comment]
I was flashing one module with the files we have used and it was working OK in ZHA.
Then paring it with Dirigera and its was start updating the firmware to 1.0.35 so i must sniffing it but if the OTA is being flashed OK we is knowing we have one working set of files

In Dirigera i only getting one remote ad can linking it to Sonoss seekers and not other extra lights or somthing and i still thing you is having strange things in the user data.

I trying getting the new OTA file from sniffing it and reporting back if it going well.

**Comment by MattWestb** [comment]
I was testing with TF GW with original 1.0.12 updated 1.0.32 on untouched remotes and also on module with updated by Dirigera to 1.0.35 (was 1.0.32 before update).
All is making one speaker that can being bound to Symfonisk or Sonos speekers.
And also one device called shortcut button that can adding single, double and long pressing for shortcut button 1 and 2.
Its also make one group but it hidden but then deleting the remote its saying it was deleting group X.

My test device is is having one user data that is not  having any data so its using the firmware default and its looks working very good then Diriger have updating it from 1.0.32 to 1.0.35. Also TF GW is not updating the remote also the 1.0.12.
I have checking with commander with blank test and the main flash is not as expected and its reporting that user data is blank.

So my conclusion is that your devices is having some strange written in the user data that is making the shortcut ad other devices and i was having the same problem before erasing the user data i was getting 2 extra lights in Dirigera from them.
The most strange is that you is also getting it with the untouched one but as i have saying the manufacture can have doing the data setting wrong in the factory posses.

Also possible is that the TF GW is having one hick up and the tested Dirigera 2 but i think its very unlikely that both is having it at the same time.

Can you trying flashing your firmware on one module and writing one empty user data on it and reading it back an look if its only `FF` in it and testing paring it with TF GW ?

**Comment by MattWestb** [comment]
Symfonisk gen 2 version `1.0.35` `ota` file for OTA upgrading and S37 for SWD flashing.
Not tested only done with Dirigera but shall working OK the the signing is OK.
[ota_t0x110e_m0x117c_v0x01000035.zip](https://github.com/basilfx/TRADFRI-Hacking/files/11157620/ota_t0x110e_m0x117c_v0x01000035.zip)

**Comment by CableCatDK** [comment]
I have dumped the userdata from an original module, and its content is byte-identical with the one we made.
<pre>
S3150FE0020053594D464F4E49534B20736F756E6420CD
S3150FE0021072656D6F74652067656E32FFFFFFFFFFD6
S3150FE00220FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE9
S3150FE00230014532313233FFFFFFFFFFFFFFFFFFFFC5
</pre>

**Comment by CableCatDK** [comment]
Dump of remote upgrade from 1.0.32 to 1.0.35.
[IKEA_E2123_1.0.35_firmware.zip](https://github.com/basilfx/TRADFRI-Hacking/files/11186000/IKEA_E2123_1.0.35_firmware.zip)

**Comment by MattWestb** [comment]
Interesting is this 2 blocks in user data:

0x242-0x243 `00 e2`
And 
0x26a-0x26b `00 00`

Both is after the custom name IKEA have putting in and is likely changing the configuration of the device.

Im still confused then my 2 remotes is working OK in TF and DG.
I trying flashing your user data an main flash tomorrow and paring and looking how its looking and reporting back.

**Comment by CableCatDK** [comment]
The name in userdata changed between the 2 versions
1.0.32: SYMFONISK sound remote 2, E2123
1.0.35: SYMFONISK sound remote gen2

**Comment by MattWestb** [comment]
Then i must dumping my test module then i was erasing the user data ans it shall being empty. Normal is user data not updated with OTA but its possible but little tricky and i have not seen IKEA doing it before.
Normally they is putting in one override name in the OTA firmware and dont need doing extra things.

**Comment by MattWestb** [comment]
TF is looping the update for Styrbar then the OTA file is having one lower number in the meta data (header in the OTA file) then its inside the file and its doing `==` and its not equal and its getting one more update next day after its have flashing the device.

I was erasing my test module that was having the user data empty and flashing your latest bin file for MF and UD and paring it with Dirigera and is getting one device that looks OK.
Then paring it with TF and its making one device as one light but with the real name is not one remote (you must have the APP in latest version or is not being shown).

Then erasing the module and writing my empty userdat bin file and doing chip erases and verifying both MF and UD is empty.  
Flashing your latest updated file 1.0.35 and paring it with TF and is getting one speaker and one device with with 2 shortcut bottoms. Paring my updated remote and its doing the same.
![Screenshot_20230410-180923_Home smart 1](https://user-images.githubusercontent.com/49618193/230946984-70057f26-052f-460e-aed3-8eea55be76f9.jpg)
You see 2 speakers and 2 shortcut buttons the module is running your firmware but with empty user data = its taking the information from the firmware like name and hardware settings and my original updated 1.0.32.

With Dirigera i only getting one speaker but i think the shortcut buttons is being for playlist in Sonos.

Try flashing this file in user data or at 0xFE00000 if cant doing regions on your test module and paring it new with TF GW and look how it working.

[IKEA_MG210L-BlankUD.zip](https://github.com/basilfx/TRADFRI-Hacking/files/11192123/IKEA_MG210L-BlankUD.zip)

PS thanks for the Styrbar with both last versions !!

PPS: Im still confused over that your original dumped user data is not working and its looks like you have writing it by mistake but i think its **not** possible then the device was debug locked.

**Comment by MattWestb** [comment]
TF have posting the last update in the feed:

- Release Version: 1.24.0 11th April 2023
  - New features and changes in Accessories:
    - SYMFONISK sound remote gen 2 (V-1.0.35)
    - Upgrade to stack v7.1

**Comment by CableCatDK** [comment]
Super. I will try and see what happes if I join an older version remote to the gateway.

**Comment by CableCatDK** [comment]
The remote did not appear in the list of device to be updated. But the app said it was updating the devices. It did finish, but the system still thinks the remote is used to control light, not speakers.

**Comment by MattWestb** [comment]
My feeling is that is somthing strange with your user data.
I was having the same then flashing my first module that was one from one light and was getting one sound remote and 2 lights and flashing one empty user data it was one sound control and 2 shortcut buttons.
I was seen your TF is in latest version so its OK (old is not supporting the device) and its making group for the remote but its being hidden in the app and then wrong device type the app is making one light group for it.

**Comment by CableCatDK** [comment]
I am thinking I will update my last untouch orginal remote in the gateway, and then dump the userdata.

**Comment by CableCatDK** [comment]
I suspect the first update from 1.0.12 to 1.0.32 by hub messed up the userdata, which I then dumped from an original remote. The update to 1.0.35 attempted to fix it, but failed. We will see when I have the userdata from a remote updated from 1.0.12 to 1.0.35 by the gateway.

**Comment by CableCatDK** [comment]
The userdata for 1.0.35 on gateway is identical to 1.0.32 on hub

1.0.12 to 1.0.35 gateway or 1.0.12 to 1.0.32 hub
<pre>
S3150FE0020053594D464F4E49534B20736F756E6420CD	"SYMFONISK sound "
S3150FE0021072656D6F74652067656E32FFFFFFFFFFD6	"remote gen2"
S3150FE00220FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE9
S3150FE00230014532313233FFFFFFFFFFFFFFFFFFFFC5	"E2123"
S3150FE00240FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFC9
S3150FE00250FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB9
S3150FE00260FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFA9
</pre>

1.0.12 to 1.0.32 to 1.0.35 hub
<pre>
S3150FE0020053594D464F4E49534B20736F756E6420CD	"SYMFONISK sound "
S3150FE0021072656D6F74652032FFFFFFFFFFFFFFFF13	"remote 2"
S3150FE00220FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE9
S3150FE00230014532313233FFFFFFFFFFFFFFFFFFFFC5	"E2123"
S3150FE00240FFFF00E2FFFFFFFFFFFFFFFFFFFFFFFFE5
S3150FE00250FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB9
S3150FE00260FFFFFFFFFFFFFFFFFFFF0000FFFFFFFFA7
</pre>

**Comment by CableCatDK** [comment]
It is good we are logging every step. I have found my mistakes:
1. The handcrafted userdata had mistakes: Wrong description, missing 00E2 and 0000.
2. The comparison of the dump userdata and handcrafted userdata 4 days ago was wrong. There are the difference described before.

So the update never change the userdata. It was me who guess wrong. My mistake only affect the gateway, not the hub.

**Comment by CableCatDK** [comment]
Me dumping the userdata:
![image](https://user-images.githubusercontent.com/13363063/231868521-21d5978f-83b2-4fa5-9b74-09bf23f17fbb.jpeg)

**Comment by MattWestb** [comment]
Looks your next device is in the pipe then its being leaked and its on the way.
https://www.reddit.com/r/tradfri/comments/13f29wy/app_mentions_new_rodret_dimmer_any_details/

Also in Austria IKEA have start updating all descriptions of all TF devices with connecting dimmer switch with RODRET instead or the OnOff dimmer switch.


How is your updated hardware of Symfonisk II working ??
Is all OK after getting the updated hardware and the firmware fixed ??

**Comment by MattWestb** [comment]
Then using one new module without any caps and resistors is nearly impossible getting button events being triggered by using one pull down resistor then its need one pull up on PA03 for scanning the button input ports.
The key function is the PA03 that is working as one trigger for scanning the other inputs for changes. If the R and C components is not working OK you is not getting any triggering then pressing the buttons then the timing of PA03 + the button input is not in sync / wrong time / DC level.
Also check the DC levels on the button inputs and PA03 then not pressed and pressed and comparing with the original PCB mounted module if you is getting the same readings.

**Comment by MattWestb** [comment]
### First version of RODRET OnOff Dimmer Switch have arrived !!
6.99 € is better then the first gen buttons.
![IMG_20230714_175318](https://github.com/basilfx/TRADFRI-Hacking/assets/49618193/3aa0d8e1-b4d1-4650-b6d3-b54063e5cf1d)
Now we must waiting @CableCatDK is baying one and using the hammer on it so we is knowing how its look inside and the PCB layout.

**Comment by MattWestb** [comment]
Sorry its no Silabs module inside they have making on compact PCB with integrated antenna  and on tiny shielded very like EEFR32MG21 chip but its looks nice and cas they was not doing the same with Symfonisk 2 and Styrbar then have making them more compact. Its also looks they is having one stepup converter so can using rechargeable batteries then it shall being tricky without it getting the radio and CPUs working stable.
![IMG_20230716_160638](https://github.com/basilfx/TRADFRI-Hacking/assets/49618193/a188f5d5-225a-4310-88dd-94c23f59161e)
![IMG_20230716_160657](https://github.com/basilfx/TRADFRI-Hacking/assets/49618193/9f05b95a-6449-4c58-9e09-bdb188dcd870)
I is missing test pas for the chip rest that can being good to have then recover chips that have getting problems but perhaps its hidden someplace.

**Comment by MattWestb** [comment]
Was trying connecting to the debug with my Silabs WSDK but it was not liking it.
Was looking  on the power and with one 1.5V battery the pads is having 2.22V so the next try was without the plus and only GND and the debug lines but still mot like talking to my.
With the old IKEA module its looks Segger have make changes and its needs the hardware rest for getting the chip in debug mode, The problem here is the we dont have the reset exposed.
In GSDK its possible using some of the debug lines for other things then booting the app and then one hardware reset is needed for hocking the chip in early stage of booting before its disabling the debug interface. I can IKEA have using that trick.

If some one have one device and one SWD debugger pleas try see if can getting contact with the chip then its more then likely on EFR32MG21 without hardware secure storage as the normal MG21 module is using but i can its  have other ports that the module but its depends of the chip version used.
I have not being so brutal as our Danish friend so no hammer for getting the chip "free" yet.

**Comment by jlunz** [comment]
@MattWestb :
You can access RESETn at the following location, outside the shield can:
![image](https://github.com/basilfx/TRADFRI-Hacking/assets/117189/0d258e0a-d0d2-48fa-809c-09975e2148fd)


I tried first with a converted [ST-Link to J-Link](https://www.segger.com/products/debug-probes/j-link/models/other-j-links/st-link-on-board/) without much success.
I then used a [Simplicity Link Debugger](https://www.silabs.com/development-tools/wireless/simplicity-link-debugger?tab=overview) and was able to read Info etc.
![image](https://github.com/basilfx/TRADFRI-Hacking/assets/117189/97df1ac0-eba8-4e65-bc07-4048cb8b7e7a)
Flashing the baremetal blink exampled worked without an issue.

**Comment by MattWestb** [comment]
Sound great and thanks @jlunz !!
So IKEA have labeling it GND for making all creative persons not  testing it !!!!
So with WSTK it was working then you also connected the chip reset ?
The good thing is looks IKEA have not debut logging your device so it shall being possible using it for future device "conversion" .
Can you reading / dumping the main flash and the user data then we can needing it for the OpenClose and shortcut buttons and perhaps is they is doing one updated valve remote.

Also most J-Link clones is not working with MG2X chips then dont  have the security implanted in the debugger and if converting it with Segger its being locked to only flashing the OME (ST) chips.
WSTK is having some magic implanted so its possible recovering MG1X chips (i gave hard soft braking 3 modules but was able recovering them with the WSTK by stalling the processor with the reset).

I shall making one test dumping the flash in the weekend and reporting back.

Do you also have "found " the pin / pad function so we can see if its possible flashing it on the normal MG21 module ?

One more thanks for the great information !!!

**Comment by jlunz** [comment]
I was not completely sure if I had unlocked debug access / erased flash with my ST-Link before, so I wired up a new remote.
It seems that the device is locked, I am getting the following output. Anything I should / can try before a erase?
```
commander device --device EFR32MG21A010F1024IM32 info
JLinkError: Could not find core in Coresight setup
 
WARNING: Could not connect to target device
ERROR: Debug access is locked. Could not connect to device
DONE
```

**Comment by MattWestb** [comment]
Servus J.

OK the chip was debug locked ;-((

Can you dumping the user data flash part ?
Its holding the hard and software configuration IKEA is using for the device and with it we can using one unlocked chip and flashing one bootloader from STYRBAR and flashing the extracted OTA file and the device shall working OK (was working with Symfonisk G2). 

I have not have time looking on my test device but i have 2 in production and looks working great but one LIDL / tuya power strip was playing nasty games and was destroying my production network by have power problems and one IKEA outlet and one GU10 CWS3 was repowering oft and was making device accouterments that is being badly handled by all silabs chips and must being repowered for working OK after that (shall being fixed in later Zigbee stacks).

Thanks for sharing your findings and good hacking for you.

**Comment by jlunz** [comment]
Ok, just to be sure:
I unlock:
```
commander device lock --debug disable
```
and then try to read userdata:
```
commander readmem --region @userdata --outfile dumpuser.hex
```

**Comment by MattWestb** [comment]
I think you can doing the unlock from GUI but i have not trying it and shall doing the same thing as the CLI.

I was not getting the region working but other users was having working OK so i was using memory addresses and its working the same way and shall being OK.

It can being that you need adding the standard `-device EFR32MG21A010F1024IM32` so commander is knowing the chip layout or its complaining.

Out file is HEX good i normally making more formats if some its not being OK and BIN is easy reading and editing but no check for errors but all is working and can being converted with commander if the in file is OK.

**Comment by jlunz** [comment]
Here we go:
[dumpuser(bin,hex,s37).zip](https://github.com/basilfx/TRADFRI-Hacking/files/12267644/dumpuser.bin.hex.s37.zip)

It has the following in:
<pre>
00000000: ffff ffff ffff ffff ffff ffff ffff ffff  ................
...
000001f0: ffff ffff ffff ffff ffff ffff ffff ffff  ................
00000200: 524f 4452 4554 2044 696d 6d65 72ff ffff  RODRET Dimmer...
00000210: ffff ffff ffff ffff ffff ffff ffff ffff  ................
00000220: ffff ffff ffff ffff ffff ffff ffff ffff  ................
00000230: 0145 3232 3031 ffff ffff ffff ffff ffff  .E2201..........
00000240: ffff 00e2 ffff ffff ffff ffff ffff ffff  ................
00000250: ffff ffff ffff ffff ffff ffff ffff ffff  ................
00000260: ffff ffff ffff ffff ffff 0000 ffff ffff  ................
00000270: ffff ffff ffff ffff ffff ffff ffff ffff  ................
...
000003f0: ffff ffff ffff ffff ffff ffff ffff ffff  ................
</pre>

<details><summary>Full output, for those wanting to do the same</summary>

```
$ commander device --device EFR32MG21A010F1024IM32 lock --debug disable
Unlocking debug access (triggers a mass erase)...
Chip successfully unlocked.
DONE

$ commander device --device EFR32MG21A010F1024IM32 info
Part Number    : EFR32MG21A010F1024IM32
Die Revision   : A1
Production Ver : 41
Flash Size     : 1024 kB
SRAM Size      : 96 kB
Unique ID      : bc026efffecdbd07
DONE

$ commander readmem --device EFR32MG21A010F1024IM32 --region @userdata --outfile dumpuser.hex
WARNING: No serial number or IP address given, cannot lock access to adapter.
Reading 1024 bytes from 0x0fe00000...
Writing to dumpuser.hex...
DONE
```
</details>

**Comment by MattWestb** [comment]
Thanks its looks very good perhaps i shall flashing it one one MG21 module but i dont knowing the pinout of the paring and the up and down buttons so not so easy testing is its working more then the name is OK red.

The interesting is the 0X24x and 0X26x that is doing the hard and software settings of the device if its being one OnOff, OpenCloce or one Shortcut button.

**Comment by jlunz** [comment]
> Thanks its looks very good perhaps i shall flashing it one one MG21 module but i dont knowing the pinout of the paring and the up and down buttons so not so easy testing is its working more then the name is OK red.
> 
You're welcome!
Pinout on the E2201 is:
Red LED: LED1 / PD00
Button "0": S1 / PA04
Button "∞": S2 / PA06
Button "1": S3 / PA03

**Comment by MattWestb** [comment]
Thank one more time i shall look if the information and doing some digging and test.
The MGM210L is having all pins / pads but the PA06 is not led to one pad so its shall working for all functions but its **not** possible pairing the module with that firmware but can perhaps being done with one other like Styrbar and then flashing only the OTA file so the NVM blocks is not erased but its little work testing if its working OK.

By the way then you have getting the blink working, then is you releasing your multi functions firmware for the device with okta button press and so on ????   ;-)

**Comment by CableCatDK** [comment]
How do you pull a button? On STYRBAR it is short to GND. But on SYMFONISK it is short to PA03.

**Comment by MattWestb** [comment]
S2 Pairing is shutting to GND but the other 2 is not GND but can being VSS after the voltage regulator then its not + from the battery. Must looking more.

**Comment by MattWestb** [comment]
I have unpacking the OTA file from TF OTA-feed and converting it to GBL, S37 and BIN if some one like trying recreating one working device by first flashing one working boot loader and then the "APP".
[E2201-OTA.zip](https://github.com/basilfx/TRADFRI-Hacking/files/12269087/E2201-OTA.zip)
Try first with one STYRBAR bootloader then its one of the latest not locked we have dumped that was working OK with Symfonisk 2.

**Comment by MattWestb** [comment]
I was testing connecting the PCB with my WSDK and also shouldering the reset cable and i was miss interpreting the commander then its writing "could not connect to target device" but after little trixing i cant getting the MAC and the secure elements stratus.
![RODRET01](https://github.com/basilfx/TRADFRI-Hacking/assets/49618193/c627baa1-bb6a-47c6-99f6-701d81bbeaf9)
That is true then the MCU is not online then the chip is debug locked and we can only talking with the security processor and getting the status from the secure elements but with that we is getting the status we like / need.
I also have long long cables so some time i must putting the SWD speed down from 8000 for getting it working but must testing if reading the memory map and flashing is working for knowing its OK speed.
Also its looks like the reset is not needed for getting this information but have not testing doing on unlock if its working OK without it and also getting it flashing working OK.

Then i have time i shall trying flashing the user data and the patched main flash and testing if i can getting it to work on one MG21 chip without IKEAs locked chip.

**Comment by CableCatDK** [comment]
@MattWestb I dumped 47 x ICC-A-1 from different devices. I might be useful.
[IKEA_ICC-1_ICC-A-1_dump.zip](https://github.com/basilfx/TRADFRI-Hacking/files/13325466/IKEA_ICC-1_ICC-A-1_dump.zip)

**Comment by MattWestb** [comment]
## Great thanks !!!
I have not time but i thinking doing one coordinator as i have doing with the first and second gen module of the Rodret remote it shall not being any large problem only finding the right pins on the chop and soldering cables. I can that the RP part is limited RF power but is the same with the modules the is not using the max for getting stability and one symmetric RF link.

Have you thinking doing one switch by cutting of the PCB and connecting your switches instead of the Silabs module ?

**Comment by CableCatDK** [comment]
> Have you thinking doing one switch by cutting of the PCB and connecting your switches instead of the Silabs module ?

I have not rally looked into the Rodret remote. Its large size only fits 1M5 FUGA modules (68mm high). For that size you really want at least 6 buttons, not only 2. So maybe one could flash to SYMFONISK firmware on it. Then make a daughterboard with 6-8 buttons. The result would be a 1M5 FUGA switch which uses 1xAAA batteri, but requires space behind it.

My current SYMFONISK to FUGA conversion uses 2xCR2032, and can be mounted directly on the wall.

Alkaline AAA: 860–1,200 mAh
CR2032: 220 mAh

**Comment by MattWestb** [comment]
Rodret PCB is having one buck converter (FCC papers is saying its 2 different types is being used) but i think its not working with CRXXXX cells and putting in one AAA is taking too much space so no go.
The module is using one different MG21 chip that is using different ports so can being its not working but i have not testing it serious only fast flashing it and its not running on the module.

**Comment by CableCatDK** [comment]
@MattWestb The firmware is very picky about module types. I bought MGM210L**A**22JNF2 instead of MGM210L**0**22JNF2, and the firmware is not working.

**Comment by MattWestb** [comment]
@jlunz Do you knowing the ports the RX and TX TP/Pads is using ?
I trying doing one UART bootloader and cooking one NCP / RCP firmware for using it as one coordinator and have 2 extra E2213 that i have sounding cables on and like trying if its working OK.
Also shall trying "converting" OnOff to ChortCut and back.

**Comment by jlunz** [comment]
@MattWestb I have made a quick protoype to use EmberZNet on a E2201, so you can check the pinout here (and maybe the software as well):
https://github.com/jlunz/silabs-firmware-builder/blob/jl/add_e2201_rodret/EmberZNet/E2201_Rodret/0001-configure-usart-vcom-for-e2201_rodret.patch

Here is the full traced pcb:
![pcb_trace_out](https://github.com/basilfx/TRADFRI-Hacking/assets/117189/b33d1748-b21d-4ef4-be1e-0138db925974)

Hope that helps!

**Comment by MattWestb** [comment]
Thanks i shall trying cooling one UART bootloader with TX C00 / RX C01 and see if i can getting it working OK then it shall being possible doing NCP and RCP / Thread firmware for it.

I have doing debug unlock on both Rodret and Somerig and then flashing Styrbars main flash for getting one "IKEA Bootloadre" and then the latest OTA and it working OK.
Only if flashing Rodret on Somrig its start sending light commands and also to group 0 but is keeping Somrig as device type so need update the user data if like getting it working 100%. I diont knowing if OTA is working on the unlocked / converted devices but Symfonisk 2 was working OK.

Thanks Julian for info i reporting back if all  is going well !!

**Comment by MattWestb** [comment]
Have one 4.4.0.0 X-modeme bootloader that looks loading OK and terminal commands looks working OK.
```
Gecko Bootloader v2.04.00
1. upload gbl
2. run
3. ebl info
BL >
Gecko Bootloader v2.04.00
1. upload gbl
2. run
3. ebl info
BL >
```
Also was putting in `Button "∞": S2 / PA06 (GPIO activation)` but have not testing if its right polarity (low or high) that can only being tested then one app is running OK or its always going into the  bootloder.
Pleas testing if its working if you have time and we can start trying cooking NCP and RCP firmware for the device.
[IKEAMG21-BL-4400.zip](https://github.com/basilfx/TRADFRI-Hacking/files/13986925/IKEAMG21-BL-4400.zip)

**Comment by MattWestb** [comment]
Fast cooked NCP 7.4.0.0 with only hardware configured (SW flow control enabled) and i can being flashed from the bootloader and is starting OK and i can controlling one IKEA GU10 WW and one parasoll re-joined OK so its working but its not god configured but as one prof of consent its OK.
[IKEAMG21-ncp-uart-4400.zip](https://github.com/basilfx/TRADFRI-Hacking/files/13987839/IKEAMG21-ncp-uart-4400.zip)

**Comment by MattWestb** [comment]
RCP 4.4.0.0 115200 baud no FC, only tested with `universal-silabs-flasher` but looks working OK also testing the bootloader button:
```
(311) PS C:\msys32\Portainer\belloows\311> universal-silabs-flasher --device com9 probe
2024-01-19 13:42:15.362 MW-E3222 universal_silabs_flasher.flasher INFO Probing ApplicationType.GECKO_BOOTLOADER at 115200 baud
2024-01-19 13:42:17.364 MW-E3222 universal_silabs_flasher.flasher INFO Probing ApplicationType.CPC at 460800 baud
2024-01-19 13:42:21.623 MW-E3222 universal_silabs_flasher.flasher INFO Probing ApplicationType.CPC at 115200 baud
2024-01-19 13:42:21.654 MW-E3222 universal_silabs_flasher.flasher INFO Detected ApplicationType.CPC, version '4.4.0' at 115200 baudrate (bootloader baudrate None)

(311) PS C:\msys32\Portainer\belloows\311> universal-silabs-flasher --device com9 probe
2024-01-19 13:43:08.282 MW-E3222 universal_silabs_flasher.flasher INFO Probing ApplicationType.GECKO_BOOTLOADER at 115200 baud
2024-01-19 13:43:09.320 MW-E3222 universal_silabs_flasher.flasher INFO Detected bootloader version '2.4.0'
2024-01-19 13:43:09.320 MW-E3222 universal_silabs_flasher.flasher INFO Detected ApplicationType.GECKO_BOOTLOADER, version '2.4.0' at 115200 baudrate (bootloader baudrate 115200)
```
[IKEAMG21-rcp-uart-4400.zip](https://github.com/basilfx/TRADFRI-Hacking/files/13989358/IKEAMG21-rcp-uart-4400.zip)
